### DISCUSSION TITLE
Cannot get an AWS Action to run in the correct region

### DISCUSSION THREAD

**Discussion by SivanMehta** [community/community#26135] [Actions]
<p>I am trying to get a proof of concept running for the new GitHub Actions API. For reference, you can see the failed run <a href="https://github.com/SivanMehta/lambda-actions-demo/runs/50027962" rel="nofollow noopener">here</a>. While the docker image seems to be building and running just fine, it’s failing at the end with this error message.</p>
<pre><code>An error occurred (ResourceNotFoundException) when calling the UpdateFunctionCode operation
Functions from 'us-west-2' are not reachable in this region ('us-east-1')
</code></pre>
<p>I suspect that this means that these jobs run from the us-east-1 region, and cannot talk to the region <em>I</em> want (us-west-2), is there anyway to specify which region I am talking to?</p>
<p>In a failed attempt to solve this region problem, I created an lambda in the us-east-1 region and <a href="https://github.com/SivanMehta/lambda-actions-demo/runs/50030519" rel="nofollow noopener">tried to upload there instead</a>, where I came across the following error:</p>
<pre><code>An error occurred (InvalidParameterValueException) when calling the UpdateFunctionCode operation:
Error occurred while GetObject. S3 Error Code: AuthorizationHeaderMalformed. S3 Error Message: The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-2'
</code></pre>
<p>I have already set IAM and other permissions correctly on the AWS side, and I can invoke the analogous commands manually with the aws cli, which this action purports to wrap. What changes can I make to my actions so that these complete successfully?</p>

**Comment by helaili** [comment, marked as answer]
<p>The _GitHub Action for Aws <em>defaults to the  <strong><em>us-east-1</em></strong> region. You migth want to use the optional environment variable  <strong><em>AWS_DEFAULT_REGION</em></strong> and set it to </em> <strong>us-west-2</strong> _ so both the action and the resource you are accessing are in the same region.  </p>
<p>You can find more information on <a href="https://github.com/actions/aws/tree/master/cli" rel="nofollow noopener">the action README file</a>. </p>

**Comment by SivanMehta** [comment]
<p>That worked! Thanks for your help!</p>
